### ISSUE TITLE
commenting out feature flag does not affect recompilation

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust#47003]
First I compile this code with `cargo +nightly build`:
```rust
#![feature(nll)]
pub fn foo() {
    let mut v = vec![1];
    v.push(v[0]);
}
```
Then I comment out the feature flag:
```rust
//#![feature(nll)]
pub fn foo() {
    let mut v = vec![1];
    v.push(v[0]);
}
```
Then I recompile and it does not fail, even though the code currently requires the feature flag to compile. I guess incremental recompilation is not detecting the change.

If I `cargo clean` before recompiling, or if I completely delete the line (thus changing line numbers for the following lines), the recompilation fails as it should.

**Comment by michaelwoerister** [comment]
Interesting. Thanks for the bug report, @tspiteri!
